src/rules/max-expects.ts

```diff
@@ -0,0 +1,89 @@
+import { AST_NODE_TYPES } from '@typescript-eslint/utils';
+import {
+  FunctionExpression,
+  createRule,
+  isExpectCall,
+  isTypeOfJestFnCall,
+} from './utils';
+
+export default createRule({
+  name: __filename,
+  meta: {
+    docs: {
+      category: 'Best Practices',
+      description: 'Enforces a maximum assertion calls in a test body',
+      recommended: false,
+    },
+    messages: {
+      exceededMaxAssertion:
+        'Too many assertion calls ({{ count }}). Maximum allowed is {{ max }}.',
+    },
+    type: 'suggestion',
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          max: {
+            type: 'integer',
+            minimum: 0,
+          },
+        },
+        additionalProperties: false,
+      },
+    ],
+  },
+  defaultOptions: [{ max: 5 }],
+  create(context, [{ max }]) {
+    let count = 0;
+
+    const onFunctionExpressionEnter = (node: FunctionExpression) => {
+      if (!node?.parent) {
+        return;
+      }
+
+      const isTestFn =
+        node.parent.type !== AST_NODE_TYPES.CallExpression ||
+        isTypeOfJestFnCall(node.parent, context, ['test']);
+
+      if (isTestFn) {
+        count = 0;
+
+        return;
+      }
+    };
+    const onFunctionExpressionExit = (node: FunctionExpression) => {
```

Comment:
We don't need this, since we're already resetting the count whenever we enter a test function and test functions cannot be nested so we don't need to track a stack.

Reply:
Fixed in https://github.com/jest-community/eslint-plugin-jest/pull/1166/commits/b9e4ca486fe43e334ba3e8a869a0d6e7aba6668b